jetty-util/src/main/java/org/eclipse/jetty/util/resource/URLResource.java

```diff
@@ -48,7 +48,7 @@
     protected URLResource(URL url, URLConnection connection)
     {
         _url = url;
-        _urlString=_url.toExternalForm();
+        _urlString=_url.toExternalForm().replaceAll("/+|\\\\+", "/");
```

Comment:
The intent, with the replaceAll(), was to normalise the URL so that
all slashes were single and forward, but I just realised that that
regex would break URLs that starts with 'http://', for example,
turning them into 'http:/' which is not correct.
Maybe we should just use replaceAll("\\\\+", "/") to replace backslashes.


Alessandro (Alex) Nistico
alienisty@gmail.com

"Any system like Eternity, which allows men to choose their own future,
will end
 by choosing safety and mediocrity, and in such a Reality the stars are out
of reach."

The End of Eternity - Isaac Asimov

On 25 March 2015 at 13:10, Greg Wilkins <notifications@github.com> wrote:

> In
> jetty-util/src/main/java/org/eclipse/jetty/util/resource/URLResource.java
> <https://github.com/eclipse/jetty.project/pull/34#discussion_r27095361>:
>
> > @@ -48,7 +48,7 @@
> >      protected URLResource(URL url, URLConnection connection)
> >      {
> >          _url = url;
> > -        _urlString=_url.toExternalForm();
> > +        _urlString=_url.toExternalForm().replaceAll("/+|\\\\+", "/");
>
> What is this replace for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipse/jetty.project/pull/34/files#r27095361>.
>